TAs/optee_ta/AuthVars/src/varmgmt.c

```diff
@@ -207,6 +211,17 @@ AuthVarInitStorage(
         if (status == TEE_ERROR_ITEM_NOT_FOUND)
         {
             DMSG("No stored variables found");
+#ifdef CFG_AUTHVARS_DEFAULT_VARS
+            status = InitializeDefaultVariables();
+            if (status != TEE_SUCCESS) {
+                EMSG("Failed to set default variables");
+                // TODO: We should set up all the variables which were just
```

Comment:
I don't understand this TODO.  We've called setvariable on these, what more is to be done?

Reply:
For some variables this should be an invariant property of the TA (unless succesfully overwritten by the user). Specifically the secure boot DB. Example:
    - Default variables: *A*, *B*, *C*, *db*, *kek*, *pk*
    - We succeed in setting *A*, *B*, and *C*, but something goes wrong when setting *db*.
    - The TA fails to start, which is fine, and UEFI will fail to load the variables and should fail to boot, all good so far.
    - When we reboot the device the Authvars TA will see *A*, *B*, and *C* present and conclude that it has already done the default variable initialization, **BUT not db, etc.** 
    We end up in the state where *db* etc. were never initialized and therefor the system will not be configured for secure boot out of the box, nor will it ever attempt to initialize them.

I was thinking of adding a new NV object to ack as a flag to indicate that all initialization had succeeded, but it might also be nice to have an option to mark a variable as "required" and always force it back into the storage at each boot.

Reply:
If this comment could be expanded with a similar explanation (or a pointer to documentation describing it) that would be preferred.  Will merge when we have this buttoned up.  This PR is a great start but I don't want to lose the guidance around what to do when things go wrong.

Reply:
Added more detail with dc0aa4d.